File: Selenium/Project/CRM_Activity6.java
package crm_project;

import org.testng.annotations.Test;
import org.testng.annotations.BeforeClass;

import static org.testng.Assert.assertEquals;
import static org.testng.Assert.assertTrue;

import java.time.Duration;

import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.chrome.ChromeDriver;
import org.openqa.selenium.support.Color;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;
import org.testng.annotations.AfterClass;

public class CRM_Activity6 {
 
//declare WebDriver 
		WebDriver driver;
	  
	  
	  @BeforeClass
	  public void setUp() {
		  //initialize webdriver
		  driver = new ChromeDriver();
		  //open the page
		  driver.get("http://alchemy.hguy.co/crm");
	  }
	 
	  @Test(priority = 1)
	  public void pageLoginTest() {
		  //find the username and type in the username
		  driver.findElement(By.id("user_name")).sendKeys("admin");
		  //find the passwrd filed and type in the passwrd
		  driver.findElement(By.id("username_password")).sendKeys("pa$$w0rd");
		  
		  
		  //find the Log in button and click it
		  driver.findElement(By.xpath("//input[@id='bigbutton']")).click();
		//Verify that the homepage has opened.
		  String homePageUrl=driver.getCurrentUrl();
		  Assert.assertTrue(homePageUrl.contains("Home"), "Login failed");
	  }
	  @Test(priority=2)
	  public void activitiesMenuTest() {
		 
		  //find the navigation menu
		
		driver.findElement(By.className("desktop-toolbar"));
		//adding wait  to load the navigation menu.
		 WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(2));
		 
		 
		 //find AND Ensure that the “Activities” menu item exists. 
		 // WebElement activitiesMenu=driver.findElement(By.xpath("//a[contains(text(),'Activities')]"));
		
		  WebElement activitiesMenu=wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//a[contains(text(),'Activities')]")));
		 assertTrue(activitiesMenu.isEnabled());
		 assertTrue(activitiesMenu.isDisplayed());
		System.out.println("Activities Menu is exist");
		    
	  }
	 
	  @AfterClass
	  public void tearDown() {
		  //quit the browser
		  driver.quit();
	  }

	

}

File: Selenium/Project/crm_activity6.py
# Import webdriver from selenium
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.color import Color
from selenium.webdriver.support import expected_conditions as EC
from selenium.webdriver.support.wait import WebDriverWait

# Start the Driver
driver=webdriver.Chrome()
 
    # Navigate to the URL
driver.get("http://alchemy.hguy.co/crm")
#find the username and type in the username
driver.find_element(By.ID,"user_name").send_keys("admin")
			  #find the passwrd filed and type in the passwrd
driver.find_element(By.ID,"username_password").send_keys("pa$$w0rd")
			  
			  
			 # find the Log in button and click it
driver.find_element(By.XPATH,"//input[@id='bigbutton']").click()
homePage=driver.find_element(By.ID,"tab0").text
if(homePage== "SUITECRM DASHBOARD"):
    print(" homepage has opened")
else:
    print("Login failed")
     #find the navigation menu
driver.fullscreen_window()
		
driver.find_element(By.CLASS_NAME,"desktop-toolbar")
		#adding wait  to load the navigation menu.
wait = WebDriverWait(driver, 2)
		 
		 
		 ##//find AND Ensure that the “Activities” menu item exists. 
		 
		
activities_menu=wait.until(EC.visibility_of_element_located((By.XPATH,"//a[contains(text(),'Activities')]")))
if activities_menu.is_enabled() and activities_menu.is_displayed():
                print("Activities menu is present")
else:
        print("Activities menu is not present") 